### DISCUSSION TITLE
plugin: remove `deploy plugin by id` command

### DISCUSSION THREAD

**Discussion by vince-fugnitto** [eclipse-theia/theia#11414] [Q&A] [answered]
**Description**

I'd like to bring up the possibility to remove the `plugin: deploy plugin by id` command as:
- there is no vscode equivalent
- the behavior is odd, and it is not clear for end-users on how it is expected to work (ex: https://github.com/eclipse-theia/theia/issues/3632) (I'm not even sure how it is expected to find the `id` to deploy nor could I get it to work)

There are better workflows already in place for the deployement of plugins:

- install from the extensions view
- `extensions: install from vsix` command

cc @msujew @tsmaeder any thoughts?

**Comment by vince-fugnitto** [comment]
This is the only documentation I could find on the command, which is pretty unclear for end-users when using the command in the application:

![image](https://user-images.githubusercontent.com/40359487/178545382-61e374f4-ff2a-470a-b5d0-ef6bc9eeaf90.png)

In addition, these aren't _ids_ as the command label suggests.

**Comment by msujew** [comment]
I actually never knew this command existed :D

Anyway, I don't really think it serves a useful purpose and the other existing install paths are way more 1. obvious and 2. straightforward without needing any URI. I'd also be in favor of removing the command.

**Comment by tsmaeder** [comment]
I have long thought that "deploying" should not be a run-time activity, but be done outside the normal lifecycle of the back end application. In that vein, good riddance, as far as I'm concerned :-) The only real application I can think of is in development, but I agree with @msujew that there are other ways to achieve the same goals.

**Comment by vince-fugnitto** [comment, marked as answer]
Thank you both for the feedback, I've opened https://github.com/eclipse-theia/theia/pull/11417 for the cleanup.
